app/routes.py

```diff
@@ -1 +1,219 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, abort, make_response, request
+import datetime as dt
+import requests
+import os
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#Helper Functions 
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} invalid"}, 400))
+        
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+def slack_bot_msg(task_title):
+
+    PATH = "https://slack.com/api/chat.postMessage"
+    SLACK_TOKEN = os.environ.get("SLACK_TOKEN")
+    my_headers = {'Authorization' : 'Bearer '+SLACK_TOKEN}
+    query_params ={
+        "channel": "task-notifications",
+        "text": (f"Someone just completed the task {task_title}.")
+    }
+
+    requests.post(PATH, params=query_params, headers=my_headers)
+
+
+#Task routes
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    sort_param=request.args.get("sort")
+    if sort_param == "asc":
+        tasks=Task.query.order_by(Task.title.asc())
+    elif sort_param == "desc":
+        tasks=Task.query.order_by(Task.title.desc())
+    else:
+        tasks=Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append({
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)
+            }), 200
+    return jsonify(tasks_response),200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+
+    if not task.goal_id:
+        return task.response_dict(), 200
+    else:
+                return jsonify({
+            "task": {
+                "id": task.task_id,
+                "goal_id": task.goal_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                }}), 200
```

Comment:
This is another candidate for a helper method by constructing the dictionary outside the route inside the Take model